tests/api_workflow/test_api_workflow_upload_dataset.py

```diff
@@ -80,4 +82,31 @@ def create_fake_video_dataset(self, n_videos=5, n_frames_per_video=10, w=32, h=3
 
     def test_upload_video_dataset_from_folder(self):
         self.create_fake_video_dataset()
-        self.api_workflow_client.upload_dataset(input=self.video_input_dir, max_workers=0)
+        self.api_workflow_client.upload_dataset(input=self.folder_path)
+
+    def test_upload_dataset_twice(self):
+        rng = np.random.default_rng(2021)
+
+        base_upload_single_image = self.api_workflow_client._upload_single_image
+
+        # Upload with some uploads failing
+        def failing_upload_sample(*args, **kwargs):
+            if rng.random() < 0.9:
+                return base_upload_single_image(*args, **kwargs)
+            else:
+                raise ValueError()
+        self.api_workflow_client._upload_single_image = failing_upload_sample
+        self.api_workflow_client.upload_dataset(input=self.folder_path)
+
+        # Ensure that not all samples were uploaded
+        samples = self.api_workflow_client.samples_api.get_samples_by_dataset_id(dataset_id="does not matter")
+        self.assertLess(len(samples), len(self.sample_names))
```

Comment:
There's a certain probability that this test fails, right? Because if all uploads are successful the `assertLess` will be wrong.

Reply:
This is prevented by seeding the RNG, see lines above.
I also tested it a few times, always 91 out of 100 samples were uploaded successfully.